tests/driver-stage2.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
```

Comment:
I have no idea why `tests/driver-stage2.sh` exits.

Reply:
Is it possible to stick to single `tests/driver.sh` for validation?

Reply:
Yes, it would be possible. Just need to plug in 1 one argument to specify which stage's executable is being used with testing.

Reply:
I have tested `tests/driver.sh` with stage 2, but I have found several problems:
1. The compilation speed is significantly slower than stage 0.
2. Stage 2 compiler would failed to compile these test suites
https://github.com/sysprog21/shecc/blob/18e005dbe21b323dc1559195388e02e7ff101bba/tests/driver.sh#L955-L985

Reply:
> Stage 2 compiler would failed to compile these test suites

I use the following test code to test shecc first.
```c
int main() 
{ 
    printf("%d %d\n", -2147483648 / 1, -2147483648 % 1); 
    return 0;  
} 
```
Using current shecc to compile the code would result in a Bus Error, and I believe the potential cause is `__stmdb` and `__ldm` in the Arm backend, which operate on memory to store or load multiple words, .

If temporarily removing these instructions from `arm-codegen.c`, shecc will compile the test code successfully.
```diff
    case OP_div:
    case OP_mod:
        ...
        interm = __r8;
        /* div/mod emulation */
        /* Preserve the values of the dividend and divisor */
-       emit(__stmdb(__AL, 1, __sp, (1 << rn) | (1 << rm)));
        /* Obtain absolute values of the dividend and divisor */
        emit(__srl_amt(__AL, 0, arith_rs, __r8, rn, 31));
        ...
        emit(__mov_r(__AL, __r9, rn));
-       emit(__ldm(__AL, 1, __sp, (1 << rn) | (1 << rm)));
        emit(__mov_r(__AL, rd, interm));
        ...
        return;
```
---

That is, the behavior of `__stmdb` and `__ldm` may lead to unaligned memory access, thereby triggering a `Bus Error`.

Therefore, I think improving the current Arm bakcend could be a solution to resolve the issue.

reference: [ARM Cortex-A Series Programmer's Guide for ARMv7-A](https://developer.arm.com/documentation/den0013/d/Porting/Alignment)
> However, load and store multiple instructions (LDM and STM) and load and store double-word (LDRD or STRD) must be aligned to at least a word boundary.

Reply:
Currently I have resolved ARM stack misalignment by ensure `__stmdb` and `__ldm` pushes 4 registers at a time to keep it aligned.

But now the RISCV backend will stuck, but I found that current implementation does not check the formerly mentioned test suite edge case scenario: `INT_MIN` could not be approximated due to missing equality check, which leads the program keep shifting the approximated value after reaching `INT_MIN` then causes the approximated value always be 0 afterwards.

By adding additional equality test while approximating nearest power of 2 value, the program would work as expected.